R/epi_slide_opt_archive.R

```diff
@@ -0,0 +1,230 @@
+  )
+  orig_group_vars <- group_vars(.x)
+  orig_drop <- .x$private$drop
+  .x %>%
+    ungroup() %>%
+    epi_slide_opt(...) %>%
+    group_by(pick(all_of(orig_group_vars)), .drop = orig_drop)
+}
+
+#' @method epi_slide_opt epi_archive
+#' @export
+epi_slide_opt.epi_archive <-
+  function(.x, .col_names, .f, ...,
+           .window_size = NULL, .align = c("right", "center", "left"),
+           .prefix = NULL, .suffix = NULL, .new_col_names = NULL,
+           .ref_time_values = NULL, .all_rows = FALSE,
+           .progress = FALSE) {
+    # Extract metadata:
+    time_type <- .x$time_type
+    epikey_names <- key_colnames(.x, exclude = c("time_value", "version"))
+    # Validation & pre-processing:
+    .align <- arg_match(.align)
+    .f_info <- upstream_slide_f_info(.f, ...)
+    .f_dots_baked <-
+      if (rlang::dots_n(...) == 0L) {
+        # Leaving `.f` unchanged slightly improves computation speed and trims
+        # debug stack traces:
+        .f
+      } else {
+        purrr::partial(.f, ...)
+      }
+    col_names_quo <- enquo(.col_names)
+    names_info <- across_ish_names_info(
+      .x$DT, time_type, col_names_quo, .f_info$namer,
+      .window_size, .align, .prefix, .suffix, .new_col_names
+    )
+    window_args <- get_before_after_from_window(.window_size, .align, time_type)
+    if (!is.null(.ref_time_values)) {
+      cli_abort("epi_slide.epi_archive does not support the `.ref_time_values` argument",
+        class = "epiprocess__epi_slide_opt_archive__ref_time_values_unsupported"
+      )
+    }
+    if (!identical(.all_rows, FALSE)) {
+      cli_abort("epi_slide.epi_archive does not support the `.all_rows` argument",
+        class = "epiprocess__epi_slide_opt_archive__all_rows_unsupported"
+      )
+    }
+    assert(
+      checkmate::check_logical(.progress, any.missing = FALSE, len = 1L, names = "unnamed"),
+      checkmate::check_string(.progress)
+    )
+    if (isTRUE(.progress)) {
+      .progress <- "Time series processed:"
+    }
+    use_progress <- !isFALSE(.progress)
+    # Perform the slide:
+    updates_grouped <- .x$DT %>%
+      as.data.frame() %>%
+      as_tibble(.name_repair = "minimal") %>%
+      # 0 rows input -> 0 rows output for any drop = FALSE groups with 0 rows, so
+      # we can just say drop = TRUE:
+      grouped_df(epikey_names, drop = TRUE)
+    if (use_progress) progress_bar_id <- cli::cli_progress_bar(.progress, total = n_groups(updates_grouped))
+    result <- updates_grouped %>%
+      group_modify(function(group_values, group_key) {
+        res <- epi_slide_opt_archive_one_epikey(
+          group_values,
+          names_info$input_col_names,
+          .f_dots_baked, .f_info$from_package, window_args$before, window_args$after, time_type,
+          names_info$output_col_names
+        )
+        if (use_progress) cli::cli_progress_update(id = progress_bar_id)
+        res
+      }) %>%
+      as.data.frame() %>% # data.table#6859
```

Comment:
This issue? looks like they fixed it. https://github.com/Rdatatable/data.table/issues/6859

I guess you're trying to avoid having to pin data.table to a recent version?

Reply:
I filed this issue as I was writing this PR.  We could do a version range requirement I guess; has this hit CRAN?

Reply:
It's not in the 1.17.0 release, will need to wait a bit.